packages/core/src/providers/openai/OpenAIProvider.caching.test.ts

```
@@ -0,0 +1,362 @@
+import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
+import { OpenAIProvider } from './OpenAIProvider.js';
+import type OpenAI from 'openai';
+import { IContent } from '../../services/history/IContent.js';
+import { SettingsService } from '../../settings/SettingsService.js';
+import { createProviderCallOptions } from '../../test-utils/providerCallOptions.js';
+
+const { mockChatCreate, mockOpenAIConstructor } = vi.hoisted(() => {
+  const chatCreate = vi.fn();
+  const constructor = vi.fn().mockImplementation(() => ({
```

Biome (2.1.2)

[error] Do not shadow the global "constructor" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)